SwEnt-Group8/Street-work-app
# Feat/training UI

**Overview**
In order to allow the user to track his progression in the different exercices, this Pull Request adds a dashboard with a "Training" tab in the Progression UI. It will be used to see the last record of the user on a specific exercise. Its purpose is also to navigate to a screen with all the details about this exercise, that should be implemented in the future.

This feature have strong connections with the "train with a friend" feature. Some changes will be needed in the progression MVVM to keep track of the records in the different exercises, and it will be the objective of a future PR. 

**Changes**
- Add a Dashboard in the Progression Screen, to either see the achievement or the record of each exercise
- Define a new enum class to represent the exercises, and use it in the UI.
- Update the corresponding tests, with the use of this Dashboard

**How to test**

1. go to the Progression Screen
2. Click on the "Training" or the "Achievement" part of the Dashboard

you should be able to see your achievement/the record of your training!

(Note: the button are not enable, because the detail screen does not exist yet. This is the reason why the color is not the same as the other button. Once enable, the color should be the same as the rest of the UI)

**Screenshot**
<img width="211" alt="image" src="https://github.com/user-attachments/assets/28bc8799-50a7-4216-b736-1abe01b87083">
